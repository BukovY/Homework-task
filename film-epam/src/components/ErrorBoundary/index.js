export { ErrorBoundary } from "./Error Boundary";