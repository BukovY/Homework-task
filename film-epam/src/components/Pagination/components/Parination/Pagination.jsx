import React from "react";
import styles from "./Pagination.module.sass";
import classNames from "classnames/bind";

export const Pagination = ({ num, isSelect, handler }) => {
  const selectPaginationPage = () => {
    handler(num);
  };
  const cx = classNames.bind(styles);
  const paginationClass = cx("pagination", { active: isSelect });
  return (
    <div className={paginationClass} onClick={selectPaginationPage}>
      {num}
    </div>
  );
};
