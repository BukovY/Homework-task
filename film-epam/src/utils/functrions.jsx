export const genresIndexToString = (arr, map) => {
  const genresToRender = [];
  for (let i of map) {
    if (arr.indexOf(i.id) !== -1) {
      genresToRender.push(i.name);
    }
  }
  return genresToRender;
};

export const minToTime = (num) => {
  let min = num % 60;
  let hour = (num - min) / 60;
  return String(hour) + ":" + String(min);
};

export const getPhotoCard = (path) => {
  return path === null || path === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w1280/${path}`;
};

export const getPeopleCard = (img) => {
  return img === null || img === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getFilmCover = (img) => {
  return img === null || img === undefined
    ? "https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getPaginationRange = (maxPage) =>
  Array(maxPage)
    .fill(1)
    .reduce((prev, next, index) => {
      return [...prev, index + 1];
    }, []);

export const getIndexLanguage = (lang) => {
  return lang === "EN" ? 0 : lang === "RU" ? 1 : 2;
};

export const matchOnlyNumber = (str) => /^\d+$/.test(str);
