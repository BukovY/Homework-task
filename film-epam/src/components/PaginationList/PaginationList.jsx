import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Pagination from "@material-ui/lab/Pagination";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    justifyContent: "center",
    margin: "10px 0",
    cursor: "pointer",
  },
}));

export const PaginationList = ({ selected, max, handler }) => {
  const classes = useStyles();
  return (
    <Pagination
      className={classes.root}
      variant="outlined"
      boundaryCount={max}
      count={max}
      defaultPage={selected}
      onChange={(e, value) => {
        handler(value);
      }}
    />
  );
};
