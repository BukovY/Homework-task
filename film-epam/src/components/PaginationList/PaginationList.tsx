import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Pagination from "@material-ui/lab/Pagination";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    justifyContent: "center",
    margin: "10px 0",
    cursor: "pointer",
    "& button": {
      color: "white",
    },
    "& .Mui-selected": {
      backgroundColor: "rgb(251 251 251 / 22%)",
    },
  },
}));

export const PaginationList = (props: {
  selected: number;
  max: number;
  handler: any;
}) => {
  const classes = useStyles();
  return (
    <Pagination
      className={classes.root}
      variant="outlined"
      boundaryCount={props.max}
      count={props.max}
      defaultPage={props.selected}
      onChange={(e, value) => {
        props.handler(value);
      }}
    />
  );
};
