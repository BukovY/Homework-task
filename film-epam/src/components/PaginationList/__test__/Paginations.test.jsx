import React from "react";
import rerender from "react-test-renderer";
import { Pagination } from "../PaginationList";
import { fireEvent, render, screen } from "@testing-library/react";
import { Pagination } from "../components/Parination";
import { wrapperProvider } from "../../../utils/functrions";

describe("Testing <Pagination/> Component", () => {
  it("Pagination view correct", () => {
    const tree = rerender
      .create(
        wrapperProvider(
          <Pagination
            handler={() => {
              return null;
            }}
            max={5}
          />
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Event testing Pagination", () => {
  it("Change pagination", () => {
    const handleClick = jest.fn();
    render(wrapperProvider(<Pagination handler={handleClick} max={5} />));
    fireEvent.click(screen.getByText(/1/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  it("Change pagination 2 level in", () => {
    const handleClick = jest.fn();
    render(wrapperProvider(<Pagination handler={handleClick} num={"4"} />));
    fireEvent.click(screen.getByText(/4/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
