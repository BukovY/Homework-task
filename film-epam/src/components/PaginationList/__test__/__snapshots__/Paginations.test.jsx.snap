// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Pagination/> Component Pagination view correct 1`] = `
<div
  className="pagination_box"
>
  <div
    className="pagination"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="pagination"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="pagination"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="pagination"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="pagination"
    onClick={[Function]}
  >
    5
  </div>
</div>
`;
